File: 0944-Delete-Columns-to-Make-Sorted/0944-Delete-Columns-to-Make-Sorted.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        count = 0
        length = len(strs[0])
        for col in range(length):
            for row in range(1, len(strs)):
                if strs[row][col] < strs[row-1][col]:
                    count+=1
                    break
        return count
# TC : O(m*n), where n is the number of strings and m is the length of the strings.
# SC : O(1), as we only use a constant amount of space regardless of the input size.

File: 0944-Delete-Columns-to-Make-Sorted/0944-Delete-Columns-to-Make-Sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int count = 0;
        int length = strs[0].size();
        for(int col = 0 ; col< length ; col++){
            for(int row = 1; row < strs.size(); row++){
                if(strs[row][col] < strs[row-1][col]){
                    count++;
                    break;
                }
            }
        }
        return count;
    }
};
